### PR TITLE
Automate Louketo releases (#615)

### PR DESCRIPTION
**Pull request by abstractj**
# Testing this PR

* Download the `*.yml` files from this PR and add it to `.github/workflows` into the fork of
Louketo
* Push the changes to `master` at your fork
* Go to https://github.com/<yourhandle>/louketo-proxy/releases
* Draft a new release
* Specify the tag version
* Publish the release

### CODE REVIEW

**Comment by stianst** [comment]
Nice. LGTM as a starting point. In time you may need to add more things to a release like publish containers, update website, etc..

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/checkout@v2
+      # Cleans up unreachable Git objects and convert a shallow clone to a regular one if needed
+      - name: Unshallow
+        run: git fetch --prune --unshallow
+      # Run tests prior to release    
+      - name: Test
+        run: make test
+      # Publish the binaries
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
```
Comment (stianst) [inline .github/workflows/release.yml:28]: Does this thing also end up creating tags and uploading bits to github releases?

Comment (abstractj) [reply .github/workflows/release.yml:28]: @stianst that tag is created by us when, after drafting a new release, we publish it. GoReleaser will be responsible to buildm publish the binaries and generate the changelog.

**Comment by abstractj** [comment]
Converted to draft because I will add documentation to the process and also record a demo.

**Comment by abstractj** [comment]
@stianst added a releases doc and also a video with a demo.

**Comment by abstractj** [comment]
@ASzc I totally understand your point, but we still have the `Makefile` which is the "command-line" way of doing this. The purpose of this PR is to reduce the complexity while doing releases, but we can improve the process and discuss the productization details later on.

Meanwhile, I'd appreciate your review and if you could give it a try and report if something is not working on your end.

**Comment by ASzc** [comment]
@abstractj GitHub doesn't seem to like it if one pushes in the workflows from outside. I tried to add in a workflow through the online editor too, and it doesn't seem to actually add the workflow. I only have the initial example workflow.

Drafting a release doesn't seem to attach binaries automatically: https://github.com/ASzc/louketo-proxy/releases/tag/test

**Comment by abstractj** [comment]
> @abstractj GitHub doesn't seem to like it if one pushes in the workflows from outside. I tried to add in a workflow through the online editor too, and it doesn't seem to actually add the workflow. I only have the initial example workflow.
> 
> Drafting a release doesn't seem to attach binaries automatically: https://github.com/ASzc/louketo-proxy/releases/tag/test

@ASzc I believe that happened because of the specified version, which is a non valid Semver. See the error [from your logs](https://github.com/ASzc/louketo-proxy/runs/734843669?check_suite_focus=true):

```
 release failed after 0.34s error=failed to parse tag test as semver: Invalid Semantic Version
```
Could you please give it another try with something like `X.Y.Z` and if that does not work we can jump in a call to check what's wrong.

File: .github/workflows/release.yml (hunk 1 above)
Comment (abstractj) [reply .github/workflows/release.yml:28]: @stianst The only manual steps you need to do is to `Draft a new release` and once you do that, you have a new tag and need to wait for the binaries to be published.

Tagging after the release is something we could improve and explore, but I don't see any other way of doing this, if we want to just press a button. Another alternative, would be to do that through the command line (more error prone).

**Comment by ASzc** [comment]
@abstractj That seems to have worked [here](https://github.com/ASzc/louketo-proxy/releases/tag/1.33.7)

**Pull request merged**